spec/models/enterprise_spec.rb

```diff
@@ -149,10 +149,10 @@
       end
 
       context "prevent an wrong instagram link pattern" do
-        let!(:e) { create(:enterprise, instagram: 'www.instagram.com/my_user') }
+        let(:e) { build(:enterprise, instagram: 'instagram.com/my_user') }
 
-        it "expects the instagram attribute to be in the correct pattern" do
-          expect(e.instagram).to eq('@my_user')
+        it "expects the instagram attribute to not be valid" do
+          expect(e).to_not be_valid
```

Comment:
Can we add more unit tests here to verify that the regular expressions work as intended?

In this case, it would be nice if it was valid and `instagram` could return `@my_user`. We probably need to modify the regex for this a bit more: `%r{\A(?:https?://)?(?:www\.)?instagram\.com/([a-zA-Z0-9._]{1,30})/?\z}`

We should test all common valid and invalid cases. Some ideas:

Valid input:

- `` empty string or `nil`
- `@user`
- `@my_www5.example`
- `http://instagram.com/user`
- `https://instagram.com/user/`
- `https://www.instagram.com/user`
- `instagram.com/user`

Invalid input:

- `@my-user`
- `https://facebook.com/user`
- `tagram.com/user`
- `https://instagram.com/user/preferences`

That should cover most common mistakes.